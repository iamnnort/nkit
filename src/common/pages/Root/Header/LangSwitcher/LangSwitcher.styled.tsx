import styled from '@common/theme/styled-components';

export const LangSwitcher = styled.div`
  display: flex;
`;

export const LangSwitcherItem = styled.button<{
  isActive: boolean;
}>`
  display: flex;
  justify-content: center;

  min-width: 50px;

  margin-left: 5px;
  padding: 5px;
  border: 1px solid ${(props) => props.theme.whiteColor};

  background: none;
  color: ${(props) => props.theme.whiteColor};
  font-size: ${(props) => props.theme.mediumFontSize};
  cursor: pointer;

  &:hover {
    background: ${(props) => props.theme.whiteColor};
    color: ${(props) => props.theme.textColor};
  }

  ${(props) =>
    props.isActive &&
    `
    background: ${props.theme.whiteColor};
    color: ${props.theme.textColor};
  `}
`;
