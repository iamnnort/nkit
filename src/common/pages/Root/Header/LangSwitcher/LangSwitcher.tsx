import * as React from 'react';
import { useTranslation } from 'react-i18next';

import { LangSwitcher, LangSwitcherItem } from './LangSwitcher.styled';

const LangSwitcherComponent = () => {
  const { i18n } = useTranslation();
  const handleSwitch = React.useCallback((lang: string) => {
    i18n.changeLanguage(lang);
  }, []);

  return (
    <LangSwitcher>
      <LangSwitcherItem onClick={handleSwitch.bind(this, 'en')} isActive={false}>
        {'en'}
      </LangSwitcherItem>
      <LangSwitcherItem onClick={handleSwitch.bind(this, 'ru')} isActive={true}>
        {'ru'}
      </LangSwitcherItem>
    </LangSwitcher>
  );
};

export default LangSwitcherComponent;
