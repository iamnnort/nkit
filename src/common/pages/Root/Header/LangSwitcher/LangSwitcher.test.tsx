import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import LangSwitcherComponent from './LangSwitcher';
import { LangSwitcherItem } from './LangSwitcher.styled';

describe('LangSwitcherComponent', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(<LangSwitcherComponent />);
  });

  it(`should not regret`, () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`should activate a button after clicked on another one`, () => {
    expect(wrapper.find(LangSwitcherItem).at(0).props().isActive).toBe(false);
    expect(wrapper.find(LangSwitcherItem).at(1).props().isActive).toBe(true);

    wrapper.find(LangSwitcherItem).at(0).simulate('click');
    wrapper = mount(<LangSwitcherComponent />);

    expect(wrapper.find(LangSwitcherItem).at(0).props().isActive).toBe(true);
    expect(wrapper.find(LangSwitcherItem).at(1).props().isActive).toBe(false);

    wrapper.find(LangSwitcherItem).at(1).simulate('click');
    wrapper = mount(<LangSwitcherComponent />);

    expect(wrapper.find(LangSwitcherItem).at(0).props().isActive).toBe(false);
    expect(wrapper.find(LangSwitcherItem).at(1).props().isActive).toBe(true);
  });
});
