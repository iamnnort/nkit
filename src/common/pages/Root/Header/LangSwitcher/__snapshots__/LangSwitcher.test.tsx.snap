// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangSwitcherComponent should not regret 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 50px;
  margin-left: 5px;
  padding: 5px;
  border: 1px solid;
  background: none;
  cursor: pointer;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 50px;
  margin-left: 5px;
  padding: 5px;
  border: 1px solid;
  background: none;
  cursor: pointer;
  background: undefined;
  color: undefined;
}

<LangSwitcherComponent>
  <styled.div>
    <div
      className="c0"
    >
      <styled.button
        isActive={false}
        onClick={[Function]}
      >
        <button
          className="c1"
          onClick={[Function]}
        >
          English
        </button>
      </styled.button>
      <styled.button
        isActive={true}
        onClick={[Function]}
      >
        <button
          className="c2"
          onClick={[Function]}
        >
          Русский
        </button>
      </styled.button>
    </div>
  </styled.div>
</LangSwitcherComponent>
`;
